src/Host.cpp

```diff
@@ -3903,6 +4014,48 @@ void Host::createMapper(const bool loadDefaultMap)
     raiseEvent(mapOpenEvent);
 }
 
+void Host::showHideOrCreateVideoPlayer()
+{
+    if (mpMedia->mpVideoPlayer) {
+        toggleVideoPlayerVisibility();
+        return;
+    }
+
+    createVideoPlayer();
+}
+
+void Host::toggleVideoPlayerVisibility()
+{
+    bool visStatus = mpMedia->mpVideoPlayer->isVisible();
+
+    if (mpMedia->mpVideoPlayer->isFloatAndDockable()) {
+        mpMedia->mpVideoPlayer->parentWidget()->setVisible(!visStatus);
+    } else {
+        mpMedia->mpVideoPlayer->setVisible(!visStatus);
+    }
+}
+
+void Host::createVideoPlayer()
+{
+    auto pMedia = mpMedia.data();
+    auto hostName(getName());
+
+    mpDockableVideoPlayerWidget = new QDockWidget(tr("Video Player - %1").arg(hostName));
+    mpDockableVideoPlayerWidget->setObjectName(qsl("dockVideoPlayer_%1").arg(hostName));
+
+    pMedia->mpVideoPlayer = new dlgVideoPlayer(mpDockableVideoPlayerWidget, this, pMedia);
+    pMedia->mpVideoPlayer->setStyleSheet(mProfileStyleSheet);
+    mpDockableVideoPlayerWidget->setWidget(pMedia->mpVideoPlayer);
+
+    mudlet::self()->addDockWidget(Qt::RightDockWidgetArea, mpDockableVideoPlayerWidget);
+    mudlet::self()->loadWindowLayout();
+
+    TEvent videoPlayerOpenEvent{};
+    videoPlayerOpenEvent.mArgumentList.append(QLatin1String("videoPlayerOpenEvent"));
```

Comment:
I'd put a `sys` on the front of that to indicate a Mudlet system event - but I am not sure whether that is consistent with any other existing events associated with the media system...
```suggestion
    videoPlayerOpenEvent.mArgumentList.append(QLatin1String("sysVideoPlayerOpenEvent"));
```